component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -276,6 +280,12 @@ private void addOrValidateCertificate(String userName, AuthenticationContext aut
         }
 
         if (isUserCertValid) {
+            try {
+                userName = UserCoreUtil.addDomainToName(userName, getUserStoreDomainName(userName, authenticationContext));
+                UserCoreUtil.setDomainInThreadLocal(getUserStoreDomainName(userName, authenticationContext));
```

Comment:
Shall we assign getUserStoreDomainName(userName, authenticationContext) to a variable and reuse as it is duplicating?

Reply:
Done